extensions/jitsi/common/jitsi.mjs

```diff
@@ -297,62 +382,113 @@ class JitsiHandler {
-      if (this.localTracks[i].getType() === type) {
-        this.localTracks[i].mute();
+    for (const track of this.localTracks) {
+      if (track.getType() === type) {
+        track.mute();
+
+        $(`#jitsi-local > .jitsi-user-${track.getType()}`).addClass(`jitsi-user-${track.getType()}--close`);
+        $(`#local${track.getType()}`).css('display', 'none');
       }
     }
   }
 
   /**
-   * Unmute the track
+   * Unmute the local track
    * @param {string} type Which type to unmute (audio|video)
    */
   async unmute(type) {
-    for (let i = 0; i < this.localTracks.length; i++) {
-      if (this.localTracks[i].getType() === type) {
-        this.localTracks[i].unmute();
+    for (const track of this.localTracks) {
+      if (track.getType() === type) {
+        track.unmute();
+
+        $(`#jitsi-local > .jitsi-user-${track.getType()}`).removeClass(`jitsi-user-${track.getType()}--close`);
+        if (track.getType() === 'video') {
+          $(`#local${track.getType()}`).css('display', 'block');
+        }
       }
     }
   }
 
   /**
    *
+   * @param selected
+   */
+  changeAudioOutput(selected) {
+    JitsiMeetJS.mediaDevices.setAudioOutputDevice(selected.value);
+  }
+
+  /**
+   * Start the local volume meter.
    */
-  switchVideo() {
-    this.isVideo = !this.isVideo;
-    if (this.localTracks[1]) {
-      this.localTracks[1].dispose();
-      this.localTracks.pop();
+  async setLocalAudioVolumeMeter() {
+    let volumeCallback;
+    try {
+      const audioStream = await navigator.mediaDevices.getUserMedia({audio: true, video: false});
+      const audioContext = new AudioContext();
+      const audioSource = audioContext.createMediaStreamSource(audioStream);
+      const analyser = audioContext.createAnalyser();
+      analyser.fftSize = 512;
+      analyser.minDecibels = -127;
+      analyser.maxDecibels = 0;
+      analyser.smoothingTimeConstant = 0.4;
+      audioSource.connect(analyser);
+      const volumes = new Uint8Array(analyser.frequencyBinCount);
+      volumeCallback = () => {
+        // If the audio track is muted, return 0.
+        if (this.localTracks.filter(track => track.getType() === 'audio' && !track.isMuted()).length === 0) {
+          this.room?.setLocalParticipantProperty('volume', '0');
+          return;
+        }
+
+        analyser.getByteFrequencyData(volumes);
+        const averageVolume = volumes.reduce((a, b) => a + b, 0) / volumes.length;
+
+        // parsed volume = round((normalize(original volume - noise threshold) * 100 [transform [0, 1] to [0, 100]]) / 25) [discretization]
+        let averageVolumeParsed = (Math.round(((averageVolume - 30) / (analyser.maxDecibels - analyser.minDecibels - 30) * 100) / 25));
+        averageVolumeParsed = Math.min(Math.max(averageVolumeParsed, 0), 4); // clip to [0, 4]
+
+        this.room?.setLocalParticipantProperty('volume', averageVolumeParsed.toString());
+
+        if (parseInt(averageVolumeParsed) >= 1) {
```

Comment:
nit/optional: We can consider making 1 a constant so that it takes lesser efforts to change it.